### PR TITLE
Update Netty proxy settings

### PR DESCRIPTION
**Pull request by jansupol**
Allow to pass headers to the proxy
Allow to set proxy using the JDK -Dhttp.proxyHost setting

Signed-off-by: jansupol <jan.supol@oracle.com>

### CODE REVIEW

**Review by senivam — APPROVED** [review]

File: connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java [hunk 1]
```diff
@@ -246,8 +256,29 @@ protected void initChannel(SocketChannel ch) throws Exception {
...
                          InetSocketAddress proxyAddr = new InetSocketAddress(u.getHost(),
                                                                              u.getPort() == -1 ? 8080 : u.getPort());
-                         p.addLast(userName == null ? new HttpProxyHandler(proxyAddr)
-                                                    : new HttpProxyHandler(proxyAddr, userName, password));
+                         ProxyHandler proxy = createProxyHandler(jerseyRequest, proxyAddr, userName, password, connectTimeout);
+                         p.addLast(proxy);
+                     } else {
+                         ProxySelector sel = ProxySelector.getDefault();
+                         for (Proxy proxy: sel.select(requestUri)) {
+                             if (proxy.type().equals(Proxy.Type.HTTP)) {
```
Comment (jbescos) [inline connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java:264]: I'm checking the class java.net.Proxy and it has a constructor where the type could be null. Does it makes sense to change that line by any of these?:
`proxy.type() == Proxy.Type.HTTP`
or
`Proxy.Type.HTTP.equals(proxy.type())`

File: connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java [hunk 2]
```diff
@@ -246,8 +256,29 @@ protected void initChannel(SocketChannel ch) throws Exception {
...
+                                 SocketAddress proxyAddress = proxy.address();
+                                 if (InetSocketAddress.class.isInstance(proxy.address())) {
+                                     InetSocketAddress proxyAddr = (InetSocketAddress) proxyAddress;
+                                     if (proxyAddr.isUnresolved()
+                                             && proxyAddr.getHostName() != null
+                                             && proxyAddr.getHostName().startsWith("http://")) {
```
Comment (jbescos) [inline connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java:270]: Is it possible that starts by https?

**Review by jbescos — APPROVED** [review]

File: connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java (hunk 2 above)
Comment (jansupol) [reply connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java:270]: No, JDK does not support HTTPS proxy

File: connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java (hunk 1 above)
Comment (jansupol) [reply connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java]: Updated

**Review by jbescos — APPROVED** [review]

**Pull request merged**